arangod/RocksDBEngine/RocksDBPrimaryIndex.cpp

```diff
@@ -825,10 +764,11 @@ Result RocksDBPrimaryIndex::insert(transaction::Methods& trx,
   rocksdb::Status s =
       mthd->Put(_cf, key.ref(), value.string(), /*assume_tracked*/ true);
   if (!s.ok()) {
-    res.reset(rocksutils::convertStatus(s, rocksutils::index));
+    Result res = rocksutils::convertStatus(s, rocksutils::index);
     addErrorMsg(res, keySlice.copyString());
```

Comment:
The string parameter of `addErrorMsg` is changed to `std::string_view` in this PR, so we can pass the key as a string_view here.
```suggestion
    addErrorMsg(res, keySlice.stringView());
```